src/screens/ListCitiesScreen/ListCitiesScreen.tsx

```diff
@@ -56,53 +67,95 @@ export const ListCitiesScreen = (
     });
   };
 
+  const toggleOverlay = () => {
+    setVisible(!visible);
+  };
+
   return (
     <SafeAreaView edges={['top', 'left', 'right']} style={styles.root}>
-      <View>
-        <StatusBar hidden />
-        <KeyboardAvoidingView
-          behavior={Platform.OS === 'ios' ? 'padding' : 'height'}>
-          <TouchableWithoutFeedback onPress={Keyboard.dismiss}>
-            <View style={styles.conditionContainer}>
-              <Text style={styles.conditionText}>Choose the city...</Text>
-              <SearchBar
-                placeholder="Type Here..."
-                onChangeText={setSearch}
-                value={search}
-                containerStyle={styles.searchContainer}
-                style={styles.search}
-                platform="android"
+      {statusApp === requestStatus.LOADING ? (
+        <View style={styles.timerContainer}>
+          <ActivityIndicator />
+        </View>
+      ) : (
+        <>
+          <View>
+            <StatusBar hidden />
+            <KeyboardAvoidingView
+              behavior={Platform.OS === 'ios' ? 'padding' : 'height'}>
+              <TouchableWithoutFeedback onPress={Keyboard.dismiss}>
+                <View style={styles.conditionContainer}>
+                  <Text style={styles.conditionText}>
+                    {t('listScreen.choice')}
+                  </Text>
+                  <TextInput
+                    placeholder={t('listScreen.input')}
+                    onChangeText={setSearch}
+                    value={search}
+                    style={styles.search}
+                  />
+                  <View style={styles.showButtonContainer}>
+                    <Icon
+                      tvParallaxProperties
+                      name={iconsName.SEARCH}
+                      type={iconsType.MATERIAL}
+                      onPress={onShowWeatherPress}
+                      color={COLORS.TEXT_COLORS.zuccini}
+                      size={36}
+                    />
+                  </View>
+                </View>
+              </TouchableWithoutFeedback>
+            </KeyboardAvoidingView>
+            <Text style={[styles.conditionText, { textAlign: 'center' }]}>
+              {t('listScreen.favorite')}
+            </Text>
+            <FlatList
+              style={styles.listContainer}
+              keyExtractor={keyExtractor}
+              data={selectedCities}
+              renderItem={({ item }) => (
+                <TouchableOpacity
+                  activeOpacity={0.5}
+                  style={styles.cityItemContainer}
+                  onPress={() => onCityItemPress(item)}>
+                  <CityItem
+                    title={item.city}
+                    id={item.id}
+                    selected={item.selected}
+                    isDefault={item.isDefault}
+                  />
+                </TouchableOpacity>
+              )}
```

Comment:
Please create `renderItem` to separate function, desctruct the props and add types to `item`